chainstate/src/detail/tests/test_framework/framework.rs

```diff
@@ -15,15 +15,156 @@
 //
 // Author(s): A. Sinitsyn, S. Tkach
 
-use crate::detail::tests::*;
 use chainstate_storage::BlockchainStorageRead;
 use common::{
     chain::{
-        block::ConsensusData, Block, GenBlock, Genesis, OutputSpentState, Transaction, TxInput,
-        TxOutput,
+        block::{timestamp::BlockTimestamp, ConsensusData},
+        config::{Builder as ChainConfigBuilder, ChainType},
+        signature::inputsig::InputWitness,
+        Block, Destination, GenBlock, Genesis, NetUpgrades, OutPointSourceId, OutputSpentState,
+        Transaction, TxInput, TxOutput,
     },
-    primitives::{Id, Idable, H256},
+    primitives::{time, Amount, Id, Idable, H256},
 };
+use crypto::random::Rng;
+
+use crate::{
+    detail::{
+        tests::{
+            create_new_outputs, produce_test_block,
+            test_framework::{BlockTestFrameworkBuilder, ProcessBlockBuilder},
+            TestBlockInfo, ERR_CREATE_BLOCK_FAIL, ERR_CREATE_TX_FAIL,
+        },
+        BlockIndex, GenBlockIndex, TimeGetter,
+    },
+    BlockError, BlockHeight, BlockSource, Chainstate, ChainstateConfig, PropertyQueryError,
+};
+
+/// The `Chainstate` wrapper that simplifies operations and checks in the tests.
+pub struct BlockTestFramework {
+    // TODO: FIXME: Private fields?
+    pub chainstate: Chainstate,
+    // TODO: FIXME: Remove?..
+    pub block_indexes: Vec<BlockIndex>,
+}
+
+impl BlockTestFramework {
+    /// Creates a new `BlockTestFramework` instance using a builder api.
+    pub fn new() -> BlockTestFrameworkBuilder {
+        BlockTestFrameworkBuilder::new()
+    }
+
+    /// Processes a new block with the parameters specified using `ProcessBlockBuilder`.
+    pub fn process_block(&mut self) -> ProcessBlockBuilder {
```

Comment:
I think the name here is somewhat misleading, since no block processing actually happens upon calling this function. Maybe `process_block_builder`, or just `block_builder`?